File: programmers/[구현]/피보나치수.py
# 피보나치 수

def solution(n):
    d = [0] * (n + 1)
    d[0], d[1] = 0, 1

    if n < 2: return d[n]

    for i in range(2, n + 1):
        d[i] = d[i - 1] + d[i - 2]

    return (d[n] % 1234567)

File: By-Swift/By-Swift/Programmers/구현/피보나치수.swift
//
//  피보나치수.swift
//  Algorithm-Swift
//
//  Created by 이원석 on 2021/04/04.
//

func solution(_ n:Int) -> Int {
    var arr = [Int](repeating: 0, count: n+1)
    arr[0] = 0
    arr[1] = 1
    
    if n < 2 { return arr[n] }
    
    for i in 2..<n+1{
        arr[i] = (arr[i-1]+arr[i-2])%1234567
    }
    
    return arr[n]
}
